src/WebJobs.Script.WebHost/HostJsonConfigProvider.cs

```diff
@@ -0,0 +1,138 @@
+﻿// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the MIT License. See License.txt in the project root for license information.
+
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Reflection;
+using Microsoft.Azure.WebJobs.Host.Config;
+using Microsoft.Azure.WebJobs.Host.Scale;
+using Microsoft.Azure.WebJobs.Hosting;
+using Microsoft.Azure.WebJobs.Script.Configuration;
+using Microsoft.Azure.WebJobs.Script.WebHost.Extensions;
+using Microsoft.Extensions.Configuration;
+using Microsoft.Extensions.DependencyInjection;
+using Microsoft.Extensions.Options;
+using Newtonsoft.Json;
+using Newtonsoft.Json.Linq;
+using Newtonsoft.Json.Serialization;
+
+namespace Microsoft.Azure.WebJobs.Script.WebHost
+{
+    public class HostJsonConfigProvider : IDisposable, IHostJsonConfigProvider
+    {
+        private readonly JsonSerializerSettings _concurrencySettings = new JsonSerializerSettings
+        {
+            ContractResolver = new DefaultContractResolver
+            {
+                NamingStrategy = new CamelCaseNamingStrategy()
+            }
+        };
+
+        private List<Type> _extensionOptionTypes;
+        private IScriptHostManager _scriptHostManager;
+        private bool _disposedValue;
+
+        public HostJsonConfigProvider(IScriptHostManager scriptHostManager)
+        {
+            _scriptHostManager = scriptHostManager;
+            _scriptHostManager.ActiveHostChanged += ActiveHostChanged;
+        }
+
+        public Dictionary<string, JObject> ExtensionOptions { get; private set; }
+
+        public JObject ConcurrencyOption { get; private set; }
+
+        public void RegisterOptionTypes(IServiceCollection services)
+        {
+            var extensionOptions = services.Where(p => p.ServiceType.IsGenericType && p.ServiceType.GetGenericTypeDefinition() == typeof(IConfigureOptions<>) && p.ImplementationFactory != null && p.ImplementationFactory.Target.ToString().StartsWith(typeof(WebJobsExtensionBuilderExtensions).FullName)).ToList();
+            _extensionOptionTypes = extensionOptions.Select(p => p.ServiceType.GetGenericArguments()[0]).ToList();
+        }
+
+        private void ActiveHostChanged(object sender, ActiveHostChangedEventArgs e)
```

Comment:
We won't have to track active host changing anymore with HostOptionsProvider a ScriptHost level service. Each time a new host is created, a new instance will be registered naturally. We only needed this when we were using a WebHost level service to access ScriptHost level data which won't be the case anymore.

Reply:
Removed.